### ISSUE TITLE
ADD or COPY a file to a directory will lead directory to losing permissions.

### ISSUE THREAD

**Issue by drpmma** [GoogleContainerTools/kaniko#721]
**Actual behavior**
Directory permission will be changed when ADD or COPY a file to it.

I ADD a test.txt to the directory /tmp, the permission is changed from 1777 to 0755. 

It's similar to #647.
- the original image's directory permission
```shell
total 56
drwxr-xr-x    2 root     root          4096 Jul 11 17:29 bin
drwxr-xr-x    5 root     root           340 Jul 23 08:32 dev
drwxr-xr-x    1 root     root          4096 Jul 23 08:32 etc
drwxr-xr-x    2 root     root          4096 Jul 11 17:29 home
drwxr-xr-x    5 root     root          4096 Jul 11 17:29 lib
drwxr-xr-x    5 root     root          4096 Jul 11 17:29 media
drwxr-xr-x    2 root     root          4096 Jul 11 17:29 mnt
drwxr-xr-x    2 root     root          4096 Jul 11 17:29 opt
dr-xr-xr-x  478 root     root             0 Jul 23 08:32 proc
drwx------    2 root     root          4096 Jul 11 17:29 root
drwxr-xr-x    2 root     root          4096 Jul 11 17:29 run
drwxr-xr-x    2 root     root          4096 Jul 11 17:29 sbin
drwxr-xr-x    2 root     root          4096 Jul 11 17:29 srv
dr-xr-xr-x   13 root     root             0 Jun 27 06:44 sys
drwxrwxrwt    2 root     root          4096 Jul 11 17:29 tmp
drwxr-xr-x    7 root     root          4096 Jul 11 17:29 usr
drwxr-xr-x   11 root     root          4096 Jul 11 17:29 var
```
- the build image's directory permission
```shell
total 56
drwxr-xr-x    2 root     root          4096 Jul 11 17:29 bin
drwxr-xr-x    5 root     root           340 Jul 23 08:33 dev
drwxr-xr-x    1 root     root          4096 Jul 23 08:33 etc
drwxr-xr-x    2 root     root          4096 Jul 11 17:29 home
drwxr-xr-x    5 root     root          4096 Jul 11 17:29 lib
drwxr-xr-x    5 root     root          4096 Jul 11 17:29 media
drwxr-xr-x    2 root     root          4096 Jul 11 17:29 mnt
drwxr-xr-x    2 root     root          4096 Jul 11 17:29 opt
dr-xr-xr-x  482 root     root             0 Jul 23 08:33 proc
drwx------    2 root     root          4096 Jul 11 17:29 root
drwxr-xr-x    2 root     root          4096 Jul 11 17:29 run
drwxr-xr-x    2 root     root          4096 Jul 11 17:29 sbin
drwxr-xr-x    2 root     root          4096 Jul 11 17:29 srv
dr-xr-xr-x   13 root     root             0 Jun 27 06:44 sys
drwxr-xr-x    1 root     root          4096 Jul 23 08:30 tmp
drwxr-xr-x    7 root     root          4096 Jul 11 17:29 usr
drwxr-xr-x   11 root     root          4096 Jul 11 17:29 var
```

**Expected behavior**
The directory permission should remain the same.

**To Reproduce**
build the Dockerfile below with kaniko

**Additional Information**
 - Dockerfile 
```Dockerfile
FROM alpine

ADD test.txt /tmp/
```

 - Kaniko Image v0.10.0

**Comment by cvgw** [comment]
I think this will also be fixed by https://github.com/GoogleContainerTools/kaniko/pull/961

```
cgwippern@cgwippern:~/kaniko-dev/issues/721$ docker run -it --rm alpine /bin/sh
/ # ls -l /
total 56
drwxr-xr-x    2 root     root          4096 Dec 19 15:20 bin
drwxr-xr-x    5 root     root           360 Jan 10 21:22 dev
drwxr-xr-x    1 root     root          4096 Jan 10 21:22 etc
drwxr-xr-x    2 root     root          4096 Dec 19 15:20 home
drwxr-xr-x    5 root     root          4096 Dec 19 15:20 lib
drwxr-xr-x    5 root     root          4096 Dec 19 15:20 media
drwxr-xr-x    2 root     root          4096 Dec 19 15:20 mnt
drwxr-xr-x    2 root     root          4096 Dec 19 15:20 opt
dr-xr-xr-x  471 root     root             0 Jan 10 21:22 proc
drwx------    1 root     root          4096 Jan 10 21:22 root
drwxr-xr-x    2 root     root          4096 Dec 19 15:20 run
drwxr-xr-x    2 root     root          4096 Dec 19 15:20 sbin
drwxr-xr-x    2 root     root          4096 Dec 19 15:20 srv
dr-xr-xr-x   13 root     root             0 Jan 10 21:22 sys
drwxrwxrwt    2 root     root          4096 Dec 19 15:20 tmp
drwxr-xr-x    7 root     root          4096 Dec 19 15:20 usr
drwxr-xr-x   12 root     root          4096 Dec 19 15:20 var
/ # exit
cgwippern@cgwippern:~/kaniko-dev/issues/721$ docker run -it --rm gcr.io/kaniko-dev/issues:721-kaniko-v1 ls -l /
total 56
drwxr-xr-x    2 root     root          4096 Dec 24 15:04 bin
drwxr-xr-x    5 root     root           360 Jan 10 21:22 dev
drwxr-xr-x    1 root     root          4096 Jan 10 21:22 etc
drwxr-xr-x    2 root     root          4096 Dec 24 15:04 home
drwxr-xr-x    5 root     root          4096 Dec 24 15:04 lib
drwxr-xr-x    5 root     root          4096 Dec 24 15:04 media
drwxr-xr-x    2 root     root          4096 Dec 24 15:04 mnt
drwxr-xr-x    2 root     root          4096 Dec 24 15:04 opt
dr-xr-xr-x  470 root     root             0 Jan 10 21:22 proc
drwx------    2 root     root          4096 Dec 24 15:04 root
drwxr-xr-x    2 root     root          4096 Dec 24 15:04 run
drwxr-xr-x    2 root     root          4096 Dec 24 15:04 sbin
drwxr-xr-x    2 root     root          4096 Dec 24 15:04 srv
dr-xr-xr-x   13 root     root             0 Jan 10 21:22 sys
drwxrwxrwt    1 root     root          4096 Jan 10 21:20 tmp
drwxr-xr-x    7 root     root          4096 Dec 24 15:04 usr
drwxr-xr-x   12 root     root          4096 Dec 24 15:04 var
```

**Comment by cvgw** [comment]
@drpmma would you mind checking with the latest version of kaniko to see if this bug is fixed. Thanks!

**Comment by tejal29** [comment]
This is fixed by #961  and manually verified. 
Please re-open if you this issue again.

Thank you!
